contracts/LibSubmarineSimple.sol

```diff
@@ -0,0 +1,171 @@
+
+    function getCommitState(bytes32 _commitId) public view returns (
+        uint96 amountRevealed,
+        uint96 amountUnlocked,
+        uint32 blockNumber,
+        uint16 txIndex
+    ) {
+        CommitData memory sesh = commitData[_commitId];
+        return (
+            sesh.amountRevealed,
+            sesh.amountUnlocked,
+            sesh.blockNumber,
+            sesh.txIndex
+        );
+    }
+
+    /////////////
+    // Setters //
+    /////////////
+
+    /**
+     * @notice Function called by the user to reveal the session.
+     * @dev warning Must be called within 256 blocks of the commit transaction to obtain the correct blockhash.
+     * @param _commitBlockNumber Block number in which the commit tx was included.
+     * @param _embeddedDAppData optional Data passed embedded within the unlock tx. This should probably be null
+     * @param _witness Witness "secret" we committed to
+     * @param _rlpUnlockTxUnsigned RLP encoded unsigned unlock transaction
+     * @param _proofBlob the Proof blob that gets passed to ethprove to verify merkle tree inclusion in a prior block.
+     */
+    function reveal(
+        uint32 _commitBlockNumber,
+        bytes _embeddedDAppData,
+        bytes32 _witness,
+        bytes _rlpUnlockTxUnsigned,
+        bytes _proofBlob
+    ) public {
+        bytes32 commitBlockHash = blockhash(_commitBlockNumber);
+        require(commitBlockHash != 0x0, "Commit Block is too old to retreive block hash (more than 256 blocks), or does not exist");
+        require(block.number.sub(_commitBlockNumber) > commitPeriodLength, "You must wait long enough to allow the committing period to end before revealing");
+
+        UnsignedTransaction memory unsignedUnlockTx = decodeUnsignedTx(_rlpUnlockTxUnsigned);
+        bytes32 unsignedUnlockTxHash = keccak256(_rlpUnlockTxUnsigned);
+
+        require(unsignedUnlockTx.nonce == 0);
+        require(unsignedUnlockTx.to == address(this));
+
+        // fullCommit = (addressA + addressC + aux(sendAmount) + dappData + w + aux(gasPrice) + aux(gasLimit))
+        bytes32 commitId = getCommitId(
+            msg.sender,
+            address(this),
+            unsignedUnlockTx.value,
+            _embeddedDAppData,
+            _witness,
+            unsignedUnlockTx.gasprice,
+            unsignedUnlockTx.startgas
+        );
+
+        require(commitData[commitId].amountRevealed == 0, "The tx is already revealed");
+
+        SignedTransaction memory provenCommitTx;
+        uint8 provenCommitTxResultValid;
+        uint256 provenTxIndex;
+        (provenCommitTxResultValid, provenTxIndex, provenCommitTx.nonce, /* gasprice */, /* startgas */, provenCommitTx.to, provenCommitTx.value, provenCommitTx.data, /* v */ , /* r */, /* s */, provenCommitTx.isContractCreation ) = txProof(commitBlockHash, _proofBlob);
+
+        // TX_PROOF_RESULT_PRESENT = 1;
+        // TX_PROOF_RESULT_ABSENT = 2;
+        require(provenCommitTxResultValid == 1, "The proof is invalid");
```

Comment:
Since we inherit from ProvethVerifier, don't we have `TX_PROOF_RESULT_PRESENT` defined in this context?

Reply:
Fixed